### ISSUE TITLE
fmt: allow to specify max line length or somehow ignore user given line breaks

### ISSUE THREAD

**Issue by sebhoss** [open-policy-agent/opa#6895]
## What is the underlying problem you're trying to solve?

I want to improve the readability of my policy tests while still formatting them with `opa fmt`. Consider the following example test:

```rego
test_pod_monitor_deny_with_unknown_label if {
  monitoring.deny["PodMonitor/some-monitor does not use any discovery labels"] with input as mocks.pod_monitor_with_labels({"prometheus.example.cluster/discovery": "enabled"})
}
```

IMHO the readability would be improved like this:

```rego
test_pod_monitor_deny_with_unknown_label if {
  monitoring.deny["PodMonitor/some-monitor does not use any discovery labels"] 
    with input as mocks.pod_monitor_with_labels({"prometheus.example.cluster/discovery": "enabled"})
}
```

or even:

```rego
test_pod_monitor_deny_with_unknown_label if {
  monitoring.deny["PodMonitor/some-monitor does not use any discovery labels"] 
    with input as mocks.pod_monitor_with_labels({
      "prometheus.example.cluster/discovery": "enabled"
    })
}
```

## Describe the ideal solution

`opa fmt` should allow me to specify a max line length or somehow allow me to retain certain line breaks.

## Describe a "Good Enough" solution

I think allowing to specify a max line length for `opa fmt` would be good enough because that should/would cause `opa fmt` to break up long lines. Somehow ignoring certain line breaks seems complicated and error prone.

## Additional Context

Multiple `with` clauses are already split across multiple lines. Maybe the formatter could check whether the first `with` is longer than max-line-length and if it is, split it across two lines?

**Comment by anderseknert** [comment]
This is one of the items included in #4508, which was originally meant to collect fmt-related issues for a larger push... which so far hasn't happened. I don't think we'll ever want to make `opa fmt` configurable, but the tool should definitey not generate lines longer than 120 characters. What you can do in the meantime, and what we do in some places in Regal, is using intermediate variables to work around it:

```rego
test_pod_monitor_deny_with_unknown_label if {
	expected_message := "PodMonitor/some-monitor does not use any discovery labels"
	mock_input := mocks.pod_monitor_with_labels({"prometheus.example.cluster/discovery": "enabled"})
	monitoring.deny[expected_message] with input as mock_input
}
```

Not ideal, but it's something.

**Comment by sebhoss** [comment]
Oh yeah that's right, thanks! I'll sub to #4508 and close this one!
